lms/static/scripts/frontend_apps/components/AutoGradingConfigurator.tsx

```diff
@@ -0,0 +1,174 @@
+import {
+  Checkbox,
+  CheckboxCheckedFilledIcon,
+  Input,
+  RadioGroup,
+} from '@hypothesis/frontend-shared';
+import { useCallback, useId } from 'preact/hooks';
+
+export type AutoGradingConfig = {
+  /** Whether auto grading is enabled or not */
+  enabled: boolean;
+
+  /**
+   * - all_or_nothing: students need to meet a minimum value, making them get
+   *                   either 0% or 100%
+   * - scaled: students may get a proportional grade based on the amount of
+   *           annotations. If requirement is 4, and they created 3, they'll
+   *           get a 75%
+   */
+  gradingType?: 'all_or_nothing' | 'scaled';
+
+  /**
+   * - cumulative: both annotations and replies will be counted together for
+   *               the grade calculation
+   * - separately: students will have different annotation and reply goals.
+   */
+  activityCalculation?: 'cumulative' | 'separately';
+
+  requiredAnnotations?: number;
+
+  requiredReplies?: number;
```

Comment:
This looks good to me, if we merge this I'll follow the same name and structure in the backend.

Reply:
This is mostly temporal, just for me to be able to define a local state that I can interact with.

We can merge as is or make adjustments, but I was expecting further discussions before going ahead with a model definition.